WordPress/WordPressTest/SiteCreation/SiteCreationNameTracksEventTests.swift

```diff
@@ -77,18 +128,37 @@ class SiteCreationNameTracksEventTests: XCTestCase {
     func testSiteNameTracksEventFiresWhenEntered() throws {
 
         // Given
-        let siteNameViewController = try siteNameViewControllerMaker()
-        let expectedEvent = WPAnalyticsEvent.enhancedSiteCreationSiteNameEntered.value
         let expectedProperty = "My Test Site"
-        let siteNameView = try XCTUnwrap(siteNameViewController.view as? SiteNameView)
+        let siteNameViewController = try mockSiteNameViewControllerMaker(expectedProperty: expectedProperty)
+        let expectedEvent = WPAnalyticsEvent.enhancedSiteCreationSiteNameEntered.value
+        siteNameViewController.loadViewIfNeeded()
 
+        guard let button = getContinueButton(from: siteNameViewController.view) else {
+            XCTFail("Unable to find continue button")
+            return
+        }
         // When
-        siteNameView.onContinue(expectedProperty)
+        button.sendActions(for: .touchUpInside)
 
         // Then
         let lastTracked = try XCTUnwrap(TestAnalyticsTracker.tracked.last)
         XCTAssertEqual(expectedEvent, lastTracked.event)
         let siteTitle = try XCTUnwrap(lastTracked.properties[siteNameEventPropertyKey] as? String)
         XCTAssertEqual(siteTitle, expectedProperty)
     }
+
+    /// Gets a reference to the `Continue` button in a `SiteNameView` instance
+    private func getContinueButton(from view: UIView) -> UIButton? {
```

Comment:
This seems like a lot of grunt work to avoid `private lazy var continueButton: UIButton` from becoming `lazy var continueButton: UIButton`. 😄

Reply:
It certainly adds a little bit of overhead, but there are two goals here: one is certainly to avoid changing production code access for testability, when possible; the other is actually simulate the scenario we want to test, and simulating the actual button tap feels a bit closer to that.

Reply:
>the other is actually simulate the scenario we want to test, and simulating the actual button tap feels a bit closer to that

I definitely like the bonus of testing views in this regard, as much as I want to cut them out of unit tests. 👍 It does increase confidence slightly.

>one is to avoid changing production code access for testability

I guess nothing's stopping someone from writing this same function in prod if they really wanted, though. 😆

Reply:
Sure, but would you say that a PR with such code in production would not smell at least a bit? 🤣

Reply:
Absolutely! 😆